nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -43,8 +43,32 @@ let
     LogsDirectoryMode = "0750";
     # Access write directories
     UMask = "0027";
+    # Capabilities
+    CapabilityBoundingSet = "";
+    # Security
+    NoNewPrivileges = true;
     # Sandboxing
+    ProtectSystem = "strict";
+    ProtectHome = true;
```

Comment:
Just one question: You didn't include `ProtectHome` in your PR to Mastodon. Could you elaborate why?

Reply:
In upstream Mastodon used home directory - `/home/mastodon`.
With `ProtectHome = true;` mastodon not working:
```
Mar 24 15:29:32 mastodon-test systemd[1]: Started mastodon-web.
Mar 24 15:29:32 mastodon-test systemd[1533]: mastodon-web.service: Changing to the requested working directory failed: Permission denied
Mar 24 15:29:32 mastodon-test systemd[1533]: mastodon-web.service: Failed at step CHDIR spawning /home/mastodon/.rbenv/shims/bundle: Permission denied
Mar 24 15:29:32 mastodon-test systemd[1]: mastodon-web.service: Main process exited, code=exited, status=200/CHDIR
Mar 24 15:29:32 mastodon-test systemd[1]: mastodon-web.service: Failed with result 'exit-code'.
Mar 24 15:29:33 mastodon-test systemd[1]: mastodon-web.service: Scheduled restart job, restart counter is at 4.
Mar 24 15:29:33 mastodon-test systemd[1]: Stopped mastodon-web.

```

Reply:
I see :+1: